video/external/src/backend.rs

```diff
@@ -25,154 +24,52 @@ enum ProxyOrStream {
-    download_sha256: &'static str,
-}
-
 /// A video backend that falls back to the software backend for most codecs,
 /// except for H.264, for which it uses an external decoder.
 pub struct ExternalVideoBackend {
     streams: SlotMap<VideoStreamHandle, ProxyOrStream>,
+    #[cfg(feature = "openh264")]
     openh264_codec: Option<OpenH264Codec>,
     software: SoftwareVideoBackend,
 }
 
 impl Default for ExternalVideoBackend {
     fn default() -> Self {
-        Self::new(None)
+        Self::new()
     }
 }
 
 impl ExternalVideoBackend {
-    fn get_openh264_data() -> Result<OpenH264Data, Box<dyn std::error::Error>> {
-        const OS: &str = std::env::consts::OS;
-        const ARCH: &str = std::env::consts::ARCH;
-
-        let local_filenames = match OS {
-            "linux" => vec!["libopenh264.so.7", "libopenh264.so.2.4.1", "libopenh264.so"],
-            // TODO: investigate other OSes
-            _ => vec![],
-        };
-
-        // Source: https://github.com/cisco/openh264/releases/tag/v2.4.1
-        let (download_filename, download_sha256) = match (OS, ARCH) {
-            ("linux", "x86") => (
-                "libopenh264-2.4.1-linux32.7.so",
-                "b7cf0e407f99056d90cbf62787a34820a7595b2129b165319d50766e00a66704",
-            ),
-            ("linux", "x86_64") => (
-                "libopenh264-2.4.1-linux64.7.so",
-                "1392d21466bc638e68151b716d5b2086d54cd812afd43253f1adb5b6e0185f51",
-            ),
-            ("linux", "arm") => (
-                "libopenh264-2.4.1-linux-arm.7.so",
-                "fd1dfb27d30bb72e903c9d2b4c650104a4369d2e7ffe8a4a533e8db2e7e9b19e",
-            ),
-            ("linux", "aarch64") => (
-                "libopenh264-2.4.1-linux-arm64.7.so",
-                "e8ea7e42855ceb4a90e7bd0b3abeba0c58b5f97166e8b0a30eefd58e099557a4",
-            ),
-            ("macos", "x86_64") => (
-                "libopenh264-2.4.1-mac-x64.dylib",
-                "cc0ba518a63791c37571f3c851f0aa03a4fbda5410acc214ecd4f24f8d1c478e",
-            ),
-            ("macos", "aarch64") => (
-                "libopenh264-2.4.1-mac-arm64.dylib",
-                "213ff93831cfa3dd6d7ad0c3a3403a6ceedf4ac1341e1278b5b869d42fefb496",
-            ),
-            ("windows", "x86") => (
-                "openh264-2.4.1-win32.dll",
-                "83270149640469c994a62cc32a6d8c0413cd7b802b7f1f2f532159f5bdc1cedd",
-            ),
-            ("windows", "x86_64") => (
-                "openh264-2.4.1-win64.dll",
-                "081b0c081480d177cbfddfbc90b1613640e702f875897b30d8de195cde73dd34",
-            ),
-            (os, arch) => return Err(format!("Unsupported OS/arch: {}/{}", os, arch).into()),
-        };
-
-        Ok(OpenH264Data {
-            local_filenames,
-            download_filename,
-            download_sha256,
-        })
+    #[cfg(feature = "openh264")]
+    pub fn load_openh264(directory: &Path) -> Result<OpenH264Codec, Box<dyn std::error::Error>> {
```

Comment:
Hmmm... Do we need this method? I think we can call `load_openh264` directly from `H264Decoder`/`OpenH264Codec`.

Reply:
The reason I put this "proxy" here, is that I don't _really_ want to make much more than `ExternalVideoBackend` public. So, _explicitly_ reaching in from `desktop/player.rs` (incl. conditional compilation) right to `openh264.rs` would seem maybe a bit strange to me...? Even though the (new) `OpenH264Codec` type (now) has to be used implicitly out there.

But maybe I'll see it differently once I get some sleep.

Reply:
I see that, that's okay of course, but it feels strange having openh264 module pub in that case. 

Although I would recommend making the OpenH264 constructor public (and only that, the rest pub(crate) if needed), as OpenH264 is a bit of an exception here—we need to download it, and make all kinds of weird things with it. So it makes sense that `ExternalVideoBackend` does not want to have to do anything with instantiating OpenH264 and just wants the instance, moving the burden of instantiating it to the caller; compared to acting as a proxy to first instantiate it, and then require to pass the instantiated object. The current flow that we have fits more to the former scenario IMO.

So I'm leaving the decision to you, just sharing my thoughts.

Reply:
Moved decoder instantiation out to caller, using the single public "constructor", and removed the proxy method from the backend implementation.